bin/reth/src/builder/mod.rs

```diff
@@ -1356,28 +1360,47 @@ pub struct NodeHandle {
 
     /// The receiver half of the channel for the consensus engine.
     /// This can be used to wait for the consensus engine to exit.
-    consensus_engine_rx: oneshot::Receiver<Result<(), BeaconConsensusEngineError>>,
+    pub consensus_engine_rx: oneshot::Receiver<Result<(), BeaconConsensusEngineError>>,
 
     /// Flag indicating whether the node should be terminated after the pipeline sync.
-    terminate: bool,
+    pub terminate: bool,
 }
 
 impl NodeHandle {
     /// Returns the [RethRpcServerHandles] for this node.
     pub fn rpc_server_handles(&self) -> &RethRpcServerHandles {
         &self.rpc_server_handles
     }
+}
 
-    /// Waits for the node to exit, if it was configured to exit.
-    pub async fn wait_for_node_exit(self) -> eyre::Result<()> {
-        self.consensus_engine_rx.await??;
+/// A Future which resolves when the node exits
+#[derive(Debug)]
+pub struct NodeExitFuture {
```

Comment:
I think for now we can move this inside the NodeHandle, replacing the existing fields